rotkehlchen/assets/utils.py

```diff
@@ -0,0 +1,37 @@
+from typing import Any, Optional, Union
+
+from rotkehlchen.errors import UnknownAsset
+from rotkehlchen.typing import ChecksumEthAddress
+
+from .asset import EthereumToken
+from .unknown_asset import UnknownEthereumToken
+
+
+def get_ethereum_token(
+        symbol: str,
+        ethereum_address: ChecksumEthAddress,
+        name: Optional[str] = None,
+        decimals: Optional[int] = None,
+) -> Union[EthereumToken, UnknownEthereumToken]:
+    """Given a token symbol and address return the <EthereumToken>, otherwise
+    an <UnknownEthereumToken>.
+    """
+    ethereum_token: Any = None
```

Comment:
I'm newbie with typing!
`ethereum_token` is referenced to `None` for avoiding an unbounded warning. However if I do the following, mypy moans (because the function return type is `Union[EthereumToken, UnknownEthereumToken]`):
```python
ethereum_token: Union[EthereumToken, UnknownEthereumToken, None]
```
I will revisit this once I amend the `try..except..else..finally` (as you suggest below).

Reply:
Some time you an use `# type: ignore` to ignore mypy if it's too annoying and there is no other way around it.

Mypy is a double edged sword since some times it feels like you are working to satisfy the tool instead of it working to help you

Reply:
But yes as you say after removing finally it will probably be easier